src/client/mod.rs

```diff
@@ -216,12 +254,29 @@ impl Client {
         })
     }
 
-    pub fn begin(&mut self, n_metrics: u64) -> io::Result<&mut Client> {
+    pub fn begin(&mut self, n_indoms: u64, n_instances: u64, n_metrics: u64) -> io::Result<&mut Client> {
```

Comment:
Agreed, but how would that solve the problem? We need to write out the full size of the MMV before registering anything.

It'll get more weird when implement higher level metrics like count vector or histogram, where the indoms and instances will be private. I think we'll have to go back to registering everything in one go like `export(...)` later on we if need it to be user friendly.